service/src/test/java/bio/terra/axonserver/app/controller/CromwellWorkflowControllerTest.java

```diff
@@ -0,0 +1,140 @@
+package bio.terra.axonserver.app.controller;
+
+import static bio.terra.axonserver.testutils.MockMvcUtils.USER_REQUEST;
+import static bio.terra.axonserver.testutils.MockMvcUtils.addAuth;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+
+import bio.terra.axonserver.service.cromwellworkflow.CromwellWorkflowService;
+import bio.terra.axonserver.testutils.BaseUnitTest;
+import io.swagger.client.model.CromwellApiLabelsResponse;
+import io.swagger.client.model.CromwellApiWorkflowIdAndStatus;
+import io.swagger.client.model.CromwellApiWorkflowMetadataResponse;
+import io.swagger.client.model.CromwellApiWorkflowQueryResponse;
+import io.swagger.client.model.CromwellApiWorkflowQueryResult;
+import java.util.UUID;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+public class CromwellWorkflowControllerTest extends BaseUnitTest {
+  @Autowired private MockMvc mockMvc;
+  @MockBean private CromwellWorkflowService cromwellWorkflowService;
+
+  private final UUID workspaceId = UUID.randomUUID();
+  private final UUID workflowId = UUID.randomUUID();
+
+  private final String CROMWELL_WORKFLOW_STATUS_PATH_FORMAT =
+      "/api/workspaces/%s/cromwell/workflows/%s/status";
+  private final String CROMWELL_WORKFLOW_LABELS_PATH_FORMAT =
+      "/api/workspaces/%s/cromwell/workflows/%s/labels";
+  private final String CROMWELL_WORKFLOW_METADATA_PATH_FORMAT =
+      "/api/workspaces/%s/cromwell/workflows/%s/metadata";
+  private final String CROMWELL_WORKFLOW_QUERY_PATH_FORMAT =
+      "/api/workspaces/%s/cromwell/workflows/query";
+
+  @Test
+  void status() throws Exception {
+    // Stub the workspace access check, and workspace id label matching.
+    Mockito.doNothing()
+        .when(cromwellWorkflowService)
+        .validateWorkspaceAccessAndWorkflowLabelMatches(
+            workflowId, workspaceId, USER_REQUEST.getToken());
+
+    // Stub the client status response.
+    Mockito.when(cromwellWorkflowService.getStatus(workflowId))
+        .thenReturn(
+            new CromwellApiWorkflowIdAndStatus().id(workflowId.toString()).status("Fake status"));
+
+    var status =
+        mockMvc
+            .perform(
+                addAuth(
+                    get(CROMWELL_WORKFLOW_STATUS_PATH_FORMAT.formatted(workspaceId, workflowId)),
+                    USER_REQUEST))
+            .andExpect(MockMvcResultMatchers.status().isOk())
+            .andReturn()
+            .getResponse();
```

Comment:
Should we verify if the response serializes properly as well in the unit tests? 

In the controller, I convert the responses from the Cromwell API model, to our own model.

Reply:
yeah, would be good to validate the response here.  not sure if that's what you mean.